packages/models-library/src/models_library/utils/pydantic_tools_extension.py

```diff
@@ -0,0 +1,13 @@
+from typing import Optional, TypeVar
+
+from pydantic import ValidationError
+from pydantic.tools import parse_obj_as
+
+T = TypeVar("T")
+
+
+def parse_obj_or_none(type_: type[T], obj) -> Optional[T]:
```

Comment:
It might be better to use functional programming patterns in these cases like `Maybe`, see example Implementation but espeically discussion in https://returns.readthedocs.io/en/latest/index.html#maybe-container

Reply:
that library phylosophy looks interesting to me! Will follow up on that thought! thx!